client/galaxy/scripts/mvc/library/library-foldertoolbar-view.js

```diff
@@ -1238,6 +1195,175 @@ var FolderToolbarView = Backbone.View.extend({
+                collection_item.id = checked_items[i];
+                collection_item.name = dataset.get("name");
+                collection_item.deleted = dataset.get("deleted");
+                collection_item.state = dataset.get("state");
+                collection_elements.push(collection_item);
+            }
+        } else if (elements_source === "folder") {
+            collection_elements = new Backbone.Collection(
+                Galaxy.libraries.folderListView.folder_container.get("folder").where({type: "file"})
+            ).toJSON();
+        }
+        let new_history_name = this.modal.$("input[name=history_name]").val();
+        if (new_history_name !== "") {
+            this.createNewHistory(new_history_name)
+                .done(new_history => {
+                    mod_toastr.success("History created");
+                    this.collectionImport(collection_elements, new_history.id, new_history.name);
+                })
+                .fail((xhr, status, error) => {
+                    mod_toastr.error("An error occurred.");
+                });
+        } else {
+            let selected_history_id = this.select_collection_history.value();
+            let selected_history_name = this.select_collection_history.text();
+            this.collectionImport(collection_elements, selected_history_id, selected_history_name);
+        }
+    },
+
+    collectionImport: function(collection_elements, history_id, history_name) {
+        let modal_title = `Creating Collection in ${history_name}`;
+        let creator_class;
+        let creationFn;
+        if (this.collectionType === "list"){
+            creator_class = LIST_CREATOR.ListCollectionCreator;
+            creationFn = (elements, name, hideSourceItems) => {
+                elements = elements.map(element => ({
+                    id: element.id,
+                    name: element.name,
+                    src: "ldda"
+                }));
+                return this.createHDCA(elements, this.collectionType, name, hideSourceItems, history_id);
+            };
+            LIST_CREATOR.collectionCreatorModal(
+            collection_elements,
+            {creationFn: creationFn, title: modal_title},
+            creator_class
+            );
+        } else if (this.collectionType === "paired"){
+            creator_class = PAIR_CREATOR.PairCollectionCreator;
+            creationFn = (elements, name, hideSourceItems) => {
+                elements = [
+                { name: "forward", src: "ldda", id: elements[0].id },
+                { name: "reverse", src: "ldda", id: elements[1].id }
+                ];
+                return this.createHDCA(elements, this.collectionType, name, hideSourceItems, history_id);
+            };
+            LIST_CREATOR.collectionCreatorModal(
+            collection_elements,
+            {creationFn: creationFn, title: modal_title},
```

Comment:
I think in each case this option array should have ``defaultHideSourceItems: true`` enabled. It'd be better to not see the original HDAs in the history by default. It makes even moderately large collections difficult to work with.

Reply:
I suspect this is overruled in the creator modal itself (which has its own checkbox for this, but I will check)

Reply:
you are right, the modal respects this and I adjusted my code to take advantage of that and hide the source items by default